src/components/dls/ModalNew/Modal.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import styles from './Modal.module.scss';
+
+const Modal = ({ children, trigger }) => (
+  <DialogPrimitive.Root>
+    <DialogPrimitive.Overlay className={styles.overlay} />
+    {trigger}
+    <Content>{children}</Content>
+  </DialogPrimitive.Root>
+);
+
+const Content = React.forwardRef<HTMLDivElement, any>(({ children, ...props }, forwardedRef) => (
+  <DialogPrimitive.Content {...props} ref={forwardedRef} className={styles.content}>
+    {children}
+  </DialogPrimitive.Content>
+));
+
+const Body = ({ children }) => <div className={styles.body}>{children}</div>;
+const Title = ({ children }) => (
+  <DialogPrimitive.Title className={styles.title}>{children}</DialogPrimitive.Title>
+);
+const Subtitle = ({ children }) => (
+  <DialogPrimitive.Description className={styles.subtitle}>{children}</DialogPrimitive.Description>
+);
+
+const Actions = ({ children }) => <div className={styles.actionsContainer}>{children}</div>;
+const Action = ({ children }) => (
+  <DialogPrimitive.Close className={styles.action}>{children}</DialogPrimitive.Close>
```

Comment:
Is close the only action we will perform? I think action should have multiple types and a callback to actually perform the action.

Reply:
good point, 
yes we should pass callback here. This way we'll be able to do many other stuff with this button.

Reply:
by `types`, do you mean we'll have type like "submit" will be a green button, and then "cancel" will be a gray button?

Reply:
You can have a look at [Antd Modal's footer](https://ant.design/components/modal/#components-modal-demo-footer). in Radix there is only `Close` action technically. So we should link it to a prop like `closable` and we have a section called footer which can be anything like how Antd does it which can include a button or a text or anything.